### PR TITLE
[Model Runner V2] Use FlashInfer sampler

### PR DESCRIPTION
**Pull request by njhill**
Only used if:
- No greedy requests
- No requests with user-provided seed
- At least one request with top-k and/or top-p
- No requests require processed logprobs

### CODE REVIEW

File: vllm/v1/sample/ops/topk_topp_sampler.py [hunk 1]
```diff
@@ -19,6 +19,45 @@
...
+            "VLLM_USE_FLASHINFER_SAMPLER=0."
+        )
+        return False
+    from vllm.v1.attention.backends.flashinfer import FlashInferBackend
+
+    capability = current_platform.get_device_capability()
+    assert capability is not None
+    if FlashInferBackend.supports_compute_capability(capability):
+        logger.info_once("Using FlashInfer for top-p & top-k sampling.", scope="global")
+        return True
```
[the remark this answers is not in the document]
Comment (njhill) [reply vllm/v1/sample/ops/topk_topp_sampler.py:71]: Pre-existing, but good point. Updated,

File: vllm/v1/worker/gpu/sample/sampler.py [hunk 2]
```diff
@@ -175,22 +186,41 @@ def sample(
...
+        use_flashinfer = self.use_flashinfer
+        if use_flashinfer:
+            top_k, top_p = self.sampling_states.get_top_k_top_p(
+                expanded_idx_mapping, idx_mapping_np
+            )
+            if (
+                (top_k is None and top_p is None)
+                or self.sampling_states.any_greedy(idx_mapping_np)
+                or self.sampling_states.any_explicit_seed(idx_mapping_np)
+            ):
```
Comment (WoosukKwon) [inline vllm/v1/worker/gpu/sample/sampler.py]: QQ: Don't we need to consider whether any request requires logprobs?

Comment (njhill) [reply vllm/v1/worker/gpu/sample/sampler.py]: Actually this is already considered, if the global logprobs mode is "processed_logits" or "processed_logprobs", flashinfer sampler is disabled globally: https://github.com/vllm-project/vllm/pull/42472/changes#diff-6047245d864bf5fd68b5b947b735beca94723bad40d20bfc0803d9b3eea5c1edR35-R36

which matches v1 behavior.

However probably better to move this to a batch-level check so that we can still use FI with this config when no requests have actuall requested logprobs. Will push an update for that.

Comment (njhill) [reply vllm/v1/worker/gpu/sample/sampler.py]: @WoosukKwon now updated

**Comment by njhill** [comment]
The CI failures on the latest run appear to be unrelated.

**Review by WoosukKwon — APPROVED** [review]
LGTM!

**Comment by njhill** [comment]
This is just blocked by strange test failure that I am investigating: https://buildkite.com/vllm/ci/builds/68083#019e5fe7-d919-4810-b24a-7bc3c1629ecb

**Comment by njhill** [comment]
Remaining failure is known-unrelated.
